File: app/routers/flights.py
from fastapi import APIRouter, Depends, Query, HTTPException, status
from sqlalchemy.ext.asyncio import AsyncSession
from datetime import date
from uuid import UUID

from app.database import get_db
from app.schemas.flight import FlightSearchResponse
from app.schemas.seat_map import SeatMapResponse, FlightSeatResponse
from app.services.flight_service import search_flights, get_seat_map

router = APIRouter(prefix="/flights", tags=["Flights"])

@router.get("/search", response_model=list[FlightSearchResponse])
async def search_for_flights(
    origin: str = Query(..., description="Origin airport code"),
    destination: str = Query(..., description="Destination airport code"),
    date: date = Query(..., description="Date of departure"),
    passengers: int = Query(1, description="Number of passengers"),
    currency: str = Query("USD", description="Currency for pricing"),
    locale: str = Query("en-US", description="Locale for display"),
    db: AsyncSession = Depends(get_db)
):
    flights = await search_flights(db, origin, destination, date, passengers, currency, locale)
    
    response = []
    for flight in flights:
        response.append(FlightSearchResponse(
            id=flight.id,
            flight_number=flight.flight_number,
            origin=flight.origin,
            destination=flight.destination,
            departure_at=flight.departure_at,
            arrival_at=flight.arrival_at,
            status=flight.status,
            available_classes=[si for si in flight.seat_inventory]
        ))
    return response

@router.get("/{flight_id}/seat-map", response_model=SeatMapResponse)
async def get_flight_seat_map(
    flight_id: UUID,
    db: AsyncSession = Depends(get_db)
):
    seat_map = await get_seat_map(db, flight_id)
    if not seat_map:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Flight not found")
    return seat_map

File: frontend/src/api/flights.ts
import { apiClient } from './client';
import type { FlightSearchResponse, SeatMapResponse, FlightDetailResponse, FlightListItem } from '../types';

export interface SearchFlightsParams {
  origin?: string;
  destination?: string;
  date?: string;
  passengers?: number;
  currency?: string;
  locale?: string;
}

export const flightsApi = {
  search: async (params: SearchFlightsParams): Promise<FlightSearchResponse[]> => {
    const res = await apiClient.get<FlightSearchResponse[]>('/flights/search', {
      params: {
        ...(params.origin ? { origin: params.origin.toUpperCase() } : {}),
        ...(params.destination ? { destination: params.destination.toUpperCase() } : {}),
        ...(params.date ? { date: params.date } : {}),
        passengers: params.passengers || 1,
        currency: params.currency || 'USD',
        locale: params.locale || 'en-US',
      },
    });
    return res.data;
  },

  getSeatMap: async (flightId: string): Promise<SeatMapResponse> => {
    const res = await apiClient.get<SeatMapResponse>(`/flights/${flightId}/seat-map`);
    return res.data;
  },

  getById: async (flightId: string): Promise<FlightDetailResponse> => {
    const res = await apiClient.get<FlightDetailResponse>(`/flights/${flightId}`);
    return res.data;
  },
};

// Named function exports for backward compatibility with pages
export const searchFlightsApi = async (params: SearchFlightsParams): Promise<FlightListItem[]> => {
  const res = await apiClient.get<FlightListItem[]>('/flights/search', {
    params: {
      ...(params.origin ? { origin: params.origin.toUpperCase() } : {}),
      ...(params.destination ? { destination: params.destination.toUpperCase() } : {}),
      ...(params.date ? { date: params.date } : {}),
      passengers: params.passengers || 1,
      currency: params.currency || 'USD',
    },
  });
  return res.data;
};

export const getFlightByIdApi = async (flightId: string): Promise<FlightDetailResponse> => {
  const res = await apiClient.get<FlightDetailResponse>(`/flights/${flightId}`);
  return res.data;
};

export const getSeatMapApi = async (flightId: string): Promise<SeatMapResponse> => {
  const res = await apiClient.get<SeatMapResponse>(`/flights/${flightId}/seat-map`);
  return res.data;
};
